tests/map/test-map.c

```diff
@@ -0,0 +1,111 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "map.h"
+
+void swap(int *x, int *y)
+{
+    int tmp = *x;
+    *x = *y;
+    *y = tmp;
+}
+
+enum { NNODES = 10000 };
+
+bool test_mix_insert()
+{
+    bool failed = false;
+
+    map_t tree = map_init(int, int, map_cmp_uint);
+
+    int key[NNODES];
+    int val[NNODES];
+
+    /*
+     *  Generate data for insertion
+     */
+    for (int i = 0; i < NNODES; i++) {
+        key[i] = i;
+        val[i] = i + 1;
+    }
+
+    /* This is not a reconmended way to randomize stuff, just a simple test */
+    /* I know MT19937 might be better */
+    srand((unsigned) time(NULL));
+    for (int i = 0; i < NNODES; i++) {
+        int pos_a = rand() % NNODES;
+        int pos_b = rand() % NNODES;
+        swap(&key[pos_a], &key[pos_b]);
+        swap(&val[pos_a], &val[pos_b]);
+    }
+
+    /* add first 1/2 items */
+    for (int i = 0; i < NNODES / 2; i++) {
+        map_iter_t my_it;
+        map_insert(tree, key + i, val + i);
+        map_find(tree, &my_it, key + i);
+        if (NULL == my_it.node) {
+            failed = true;
+        }
+        // assert((*(int*) (my_it.node->data)) == val[i]);
```

Comment:
Don't leave the dead code. For testing purpose, more assertions are pretty well.